src/dswx_sar/pre_processing.py

```diff
@@ -276,59 +371,80 @@ def run(cfg):
+        'landcover': 'interpolated_landcover.tif',
+        'reference_water': 'interpolated_wbd.tif',
+    }
+
+    input_ancillary_filename_set = {
+        'dem': dem_file,
+        'hand': hand_file,
+        'landcover': landcover_file,
+        'reference_water': wbd_file,
+    }
+
+    landcover_label = get_label_landcover_esa_10()
+
+    for anc_type, anc_filename in relocated_ancillary_filename_set.items():
+        input_anc_path = input_ancillary_filename_set[anc_type]
+        ancillary_path = Path(
+            os.path.join(scratch_dir, anc_filename))
+
+        # Check if input ancillary data is valid. 
+        if not os.path.isfile(dem_file):
+            err_msg = f'Input {anc_type} file not found'
+            raise FileNotFoundError(err_msg)
+
+        if anc_type in ['reference_water']:
+            no_data = ref_water_no_data
+        elif anc_type in ['landcover']:
+            no_data = landcover_label['No_data']
         else:
-            raise FileNotFoundError
-
-    # check if interpolated DEM has valid values
-    if not dem_reprocessing_flag:
-        interpolated_dem_path = os.path.join(
-            scratch_dir, 'interpolated_DEM.tif')
-        validate_gtiff(interpolated_dem_path, [0])
-
-    # check if the interpolated landcover exists
-    landcover_interpolated_path = \
-        Path(scratch_dir) / 'interpolated_landcover.tif'
-    if not landcover_interpolated_path.is_file():
-        ancillary_reloc.relocate(landcover_file,
-                                 'interpolated_landcover.tif',
-                                 method='near')
-
-    # Check if the interpolated HAND exists
-    hand_interpolated_path = os.path.join(
-        scratch_dir, 'interpolated_hand.tif')
-    if not os.path.isfile(hand_interpolated_path):
-        # Check if compuated HAND exists
-        if hand_file is None:
-            logger.info('>> HAND file is not found, so will be computed.')
-            # hand_calc.hand(dem_interpolated_path, args.scratch_dir)
-            # args.hand_file = os.path.join(args.scratch_dir, 'temp_hand.tif')
-
-        ancillary_reloc.relocate(hand_file,
-                                 'interpolated_hand.tif',
-                                 method='near')
-    del ancillary_reloc
+            no_data = np.nan
+
+        # crop or relocate ancillary images to fit the reference 
+        # intensity (RTC) image. 
+        if not ancillary_path.is_file():
+            logger.info(f'Relocated {anc_type} file will be created from ' \
+                        f'{input_anc_path}.')
+            ancillary_reloc.relocate(
+                input_anc_path,
+                anc_filename,
+                method='near',
+                no_data=no_data)
+
+        # check if relocated ancillaries are filled with invalid values
+        validation_result = validate_gtiff(
+            os.path.join(scratch_dir, anc_filename),
+            [no_data, np.inf])
+
+        if validation_result not in 'okay':
```

Comment:
```suggestion
        if validation_result not in ['okay']:
```
I think you want to compare `validation_result` with `'okay'`. That being case, why don't we just use ` validation_result != 'okay'`?